DED-138 Tag thread and untag thread descriptions added (#1146)
livechat/livechat-public-docs: src/pages/messaging/agent-chat-api/index.mdx

### Tag Thread

Tags a given **thread**. 

> [Read how to list all available chats](https://my.livechatinc.com/settings/tags) and [create a new tag](/management/configuration-api/v2.0#add-a-tag). **Tags** are case sensitive in this method.

Author: 'create a new tag' link is not linking to the appropriate part...I have checked it and it is correct, so I do not why it is happening :/

> [Read how to list all available chats](https://my.livechatinc.com/settings/tags) and [create a new tag](/management/configuration-api/v2.0#add-a-tag). **Tags** are case sensitive in this method.

Reviewer: ```suggestion
Read how to [list all available chats](management/configuration-api/v2.0/#list-all-tags) and [create a new tag](/management/configuration-api/v2.0/#add-a-tag). **Tags** are case sensitive in this method.
```

> Read how to [list all available chats](/management/configuration-api/v2.0#list-all-tags) and [create a new tag](/management/configuration-api/v2.0#add-a-tag). **Tags** are case sensitive in this method.

Reviewer: ```suggestion
Read how to [list all available tags](/management/configuration-api/v2.0#list-all-tags) and [create a new tag](/management/configuration-api/v2.0#add-a-tag). **Tags** are case sensitive in this method.
```

Reviewer: The same should be applied in other places

> Read how to [list all available chats](/management/configuration-api/v2.0#list-all-tags) and [create a new tag](/management/configuration-api/v2.0#add-a-tag). **Tags** are case sensitive in this method.

Reviewer: How about we move the info about case sensitivity to the Request table to comply with the format proposed in [this PR](https://github.com/livechat/livechat-public-docs/pull/1145/files#diff-26505ab33f8366eff9fa237c91538cc37a2f61c11cdd44e89ba26aeb8779d7b5R2492)?
<img width="648" alt="image" src="https://user-images.githubusercontent.com/42515529/167863750-44565220-2305-4940-87f2-83465879551b.png">

Author: Yes, I like the idea! Just to confirm - all the places, including v3.5?

> To make a request, you need to specify the **tag name**, the **Id of the chat** and the **Id of the thread** you want to add a **tag** to. Active and inactive chats can be tagged, and one or more tags can be added to the specific **thread**. To tag a chat, the **tag** needs to be created first. It's possible to do that in the [LiveChat Application](https://accounts.livechat.com/) or via the [Configuration API](/management/configuration-api). Tags are case sensitive and are defined by each client individually. The **list of tags** can be found in the [LiveChat Application](https://my.livechatinc.com/settings/tags). 

Reviewer: ```suggestion
Active and inactive chats can be tagged, and one or more tags can be added to the specific **thread**. To tag a chat, the **tag** needs to be created first. It's possible to do that in the [LiveChat Application](https://accounts.livechat.com/) or via the [Configuration API](/management/configuration-api). Tags are case sensitive and are defined by each client individually. The **list of tags** can be found in the [LiveChat Application](https://my.livechatinc.com/settings/tags). 
```

Reviewer: We could shorten the description by deleting the first sentence. The same information is presented in the table below (request parameters). I suggest starting the description with a simple statement, for example:

> Tags a given thread.
>
> The rest of the description.

### Tag Thread

> Tags a given thread. The following restrictions apply:

Reviewer 2: I'm wondering whether we should call the below list "restrictions" as they're more "possibilities"? 🤔 E.g. for the threads, there's rather no restrictions as a thread can be pretty much only either active or inactive :D, and another: tags are case sensitive is not rather a restriction but a fact, you know what I mean :D

Author: The word 'restrictions' I based on the fact that that is how the intro sentences to the different bullet lists were introduced. They did not have the same info. However, some were similar https://developers.livechat.com/docs/messaging/agent-chat-api/rtm-reference#chat-users
However, I get your point, so maybe - ...Attention should be paid to the following information."?

Reviewer: What about "rules"?

Author: With 'rules', I think I have similar worries like Oliwia above with 'restrictions' - '...Active and inactive chats can be tagged..' or '...The tags’ list can be found in the LiveChat  Application.' - it more sounds like a fact/info. 
So maybe 
"...Attention should be paid to the following rules and information:..."?
or
"The following rules and information apply..." (If we want to follow the rest of the bullet lists' intros)

I cannot come up with one word describing all the points...

Reviewer 3: "terms and conditions apply" 😂 :trollface: 

Can we drop this "preface" altogether and just do a new "paragraph" with all this additional info?

Author: I would stick to the bullet list as it will be easier for the reader to go through all this info. Also, each point is different, so I believe a bullet list would be more effective.

Reviewer 3: still, some points are related and can be grouped as separate "paragraphs"

Reviewer: Plus we can replace "restrictions" with "info" if needed.

Author: Yes, I was thinking to change 'restrictions' for 'infrmation'

Reviewer: It looks like the description will shrink after all the corrections :) Let's see if it still makes sense to keep the bullet list. If there are at least 3 bullet points, then I think it's worth it. If fewer, then it's better to format it as a paragraph.

Author: Both methods will have less than 3 bullet points, so it makes no sense to keep the bullet list.  Paragraphs will be better in that case.

> - Active and inactive chats can be tagged.

Reviewer 3: let's drop this
you tag threads, not tags (just as the method name suggests 😉 )
and threads can be either active or not, there is no in-between, it's just like saying "you can tag all threads"

Reviewer: chats -> threads 👍🏻
As for active and inactive, I specifically remember @bkubiak mentioning it as a valuable addition to this description

Author: For me, it is not as obvious as it may seem to be, I would stick to this info...I believe that with inactive threads it may not be so obvious for the user that he/she can also tag them - I would suggest changing to: "Active and inactive threads can be tagged" or "You can tag active and inactive threads"

Reviewer 3: idk about @bkubiak's opinion but maybe you should add him to this review instead :trollface: 
for me this states "any thread" in a different way, and most of these methods are usable on any thread unless specified otherwise

Author: I don't know.... 😅  I still support my opinion that for some people possibility of adding tags to inactive threads may not be so obvious. 
@ale-kacperczyk WDYT? Maybe I just measure people's intelligence too low

Reviewer: I wanted to add him for counter-balance but he's not available :) 

Jokes aside, I would continue to argue that this would be a helpful addition to novice developers, but this 👇 forces me to capitulate :)

> most of these methods are usable on any thread unless specified otherwise

@kamilatokarska, let's drop this line. Despite our good intentions, keeping the info could create some potential confusion as to why for some methods we specify that it works for active and inactive threads, and for other methods, we don't.

> - One or more tags can be added to the specific **thread**.

Reviewer 3: this method allows adding a single tag in a single call so maybe it's better to emphasize this as a thread property? something like "a thread can have multiple tags"?

Reviewer 3: one more thing, we should update "chat" data structure and add "tags" array to the thread there

Author: If I understand correctly - What do you think to write it directly - One tag can be added in one call, but one thread can have multiple tags'? 🤔 

"chat" data structure and add "tags" array to the thread there - I will create a task for that in Jira and investigate 😄

Reviewer 3: no need for a separate task 😉 just add it [here](https://github.com/livechat/livechat-public-docs/blob/master/payloads/messaging/v3.4/agent-chat-api/other-structures/chat.json) cc @ale-kacperczyk 

>One tag can be added in one call, but one thread can have multiple tags

idk, the interface kinda forces you to provide a single tag, and I don't like stating obvious things :P

Reviewer: 1. It's a dilemma between "Write it explicitly in the description so that people know right away" and "Give people a minute to analyze request parameters so that they figure out what can be sent themselves".

> One tag can be added in one call, but one thread can have multiple tags

"One tag can be added in one call" can be concluded from the request parameters, so we would be restating what's already expressed in the table. That leaves us with the information that "a thread can have multiple tags", which is true, but would be more fitting for a general overview of tags (like the PR @gluszczyk prepared for v3.5). It's only loosely related to this method.

Taking all into account, I think that maybe it's best to remove this line altogether?

2. We will update the chat data structure in a separate PR right after we're done with this one. That's just a more convenient way to do it for us.

Author: Yes, I agree that if someone analyzes the request parameters then the fact that 'one tag can be added in one call' can be figured out. 
I would say that the info that 'a thread can have multiple tags' is important, but you may be right that it fits more the general overview of tag.
I will remove that info completely:)

> - To tag a chat, the **tag** needs to be created first. It can be done in the [LiveChat Application](https://accounts.livechat.com/) or via the [Configuration API](/management/configuration-api).

Reviewer: ```suggestion
- To tag a chat, the **tag** needs to be created first. It can be done in the [LiveChat Application](https://accounts.livechat.com/) or via the [Configuration API](/management/configuration-api/v2.0#tags).
```

> - To tag a chat, the **tag** needs to be created first. It can be done in the [LiveChat Application](https://accounts.livechat.com/) or via the [Configuration API](/management/configuration-api).

Reviewer 2: ```suggestion
- To tag a chat, the **tag** needs to be created first. It can be done in the [LiveChat Application](https://my.livechatinc.com/) or via the [Configuration API](/management/configuration-api).
```

Reviewer 2: Or have I missed something and we redirect to the Accounts panel instead of the AA @ale-kacperczyk? 😅

Reviewer 3: livechatINC ? someone needs "the talk" :trollface: 
anyway, I would avoid pushing the user towards AA when they are browsing API docs 😉

Reviewer: Well, actually, https://my.livechatinc.com/ still has "inc". Joke's on you XD

Reviewer 3: I didn't even consider that I need to verify it, and yet you are right, AA still uses "inc", good grief, "kajam się" XD

Reviewer 3: but still, I don't think this is the place to mention AA; you check how to tag a thread with API, and to add a new tag we send you to this GUI app... meh

Reviewer: Since we agreed to merge two points into one, this entire point can be deleted.

Author: Already deleted:)

> - To tag a chat, the **tag** needs to be created first. It can be done in the [LiveChat Application](https://accounts.livechat.com/) or via the [Configuration API](/management/configuration-api).

Reviewer 3: livechatINC ? someone needs "the talk" :trollface: 
anyway, I would avoid pushing the user towards an app when they are browsing API docs 😉

> - To tag a chat, the **tag** needs to be created first. It can be done in the [LiveChat Application](https://accounts.livechat.com/) or via the [Configuration API](/management/configuration-api).

Reviewer 3: livechatINC ? someone needs "the talk" :trollface: 
anyway, I would avoid pushing the user towards an app when they are browsing API docs 😉

> - Tags are case sensitive.

Reviewer 3: It's ok here but I would consider skipping it in 3.5

Author: Just to make sure if I get it right - you mean skipping both points completely?
To confirm - tags will not be case sensitive in 3.5, yes?

And also...I cannot find this comment but you wrote that you would avoid pushing the user toward AA when they are browsing API docs - Can I ask why cause I was not aware of that 😅

Reviewer 3: this comment was only about case sensitivity... and in 3.5 it's more complicated, there is another PR that describes changes there so maybe the description in 3.5 should be based on that

as for AA aversion... if a developer comes to create their own integration and reads API docs we should show him how to do things API-way, not asking him to use "some GUI app" to skip API requests

Reviewer: Concerning case sensitivity:

1. Keep the "- Tags are case sensitive." sentence in v3.2-v3.4.
2. As for v3.5, my assumption is that the v3.5 Tag Thread method checks if a tag already exists before tagging a thread, so we should adjust the v3.5 description. Something like, "In this method, letter case needs to be respected for tag names." This would suggest that it's only for this method, and not all tag-related methods in v3.5. 
@gluszczyk, correct me if my story's made up :)

Author: I have pushed new changes with the v3.5 as I think it will be better to discuss when seeing it:)

Reviewer 3: @ale-kacperczyk in v3.5, **Create Tag** is the first method that ignores letter case when checking if the tag already exists, and the sentence "Tags are case sensitive" becomes **mostly** true 😉 so it might be better to state case-sensitivity explicitly in tag methods

maybe the sentence can explicitly state which field needs to respect the letter case?
>In this method, `tag` is case sensitive.

Grammarly fixed case sensitivity into case-sensitivity 😉 so what's with this BrE vs AmE you talked about in the other PR? 🤔

Reviewer: @gluszczyk It may be due to your Grammarly settings (maybe, idk). I had consulted "case sensitive vs case-sensitive" with our in-house editor and he recommended to stick to "case sensitive" without a hyphen.

> - Tags are defined by each client individually. The **list of tags** can be found in the [LiveChat Application](https://my.livechatinc.com/settings/tags). 

Reviewer: ```suggestion
- Tags are defined by each customer individually. The **list of tags** can be found in the [LiveChat Application](https://my.livechatinc.com/settings/tags). 
```

Reviewer: Or maybe we should write "user"? WDYT, @oliwiapolec ?

Reviewer 2: From the mentioned I think we should lean towards "user" (or maybe license, or admin? as only admins can alter the tags), we very rarely use "client" in the copy and the "customer" would be rather connoting the end-user

Author: So I think, in that case, I would go with 'admit' if only they could alter the tags. WDYT @ale-kacperczyk?

Reviewer: My only objection to "admin" is that nowhere else are we so specific with roles in method descriptions. But 'user' might confuse readers to think that tags cannot be globally set by a license, but need to be created by each user. So, let's go with 'admin' and see what a reviewer from the API team says :)

Reviewer 3: I would combine point starting with "To tag a chat..." with this one and go with
"you can list available tags here [link]... to create a new tag check/see [link]"
so there would be no mention of the user (and in case a "user" has to be mentioned then it should be admin, as only admins have sso-scopes that allow creating tags)

also pls drop
>To tag a chat, the **tag** needs to be created first.

this is like saying you need to create a thread first, otherwise, what will you tag? kinda obvious 😉

Reviewer: It is obvious when you know it, but there are people who will try to use this method and forget that they need to create a tag first. How about:

> To tag a chat, the tag needs to be [created](/management/configuration-api/v3.5#create-tag) first.

This way, we will kill two birds with one stone (reminding people that they need to create a tag first & redirect them to the new tag methods).

Reviewer 3: that's why I wanted to combine listing and creating tags in a single point... as for "tag needs to be created first", it's not an API-obvious but a life-obvious kinda thing

Reviewer: You would think that it's life-obvious that you first need to register a property and only then are you able to update it, yet some users still try to update a non-existent property and look for property creation options in the Agent Chat API. The same might be true for tags and linking the method that allows for creating would help tackle that.

Okay, but I think merging the info about available tags & how to create tags into a single point is a nice compromise. 

In v3.2-v3.4, I would add:

Read how to [list all available tags](/management/configuration-api/v2.0#list-all-tags) and [create a new tag](/management/configuration-api/v2.0/#add-a-tag).

In v3.5, I would add a similar note but link to v3.5 methods instead.

Author: I would go with the compromise as I am also not so sure if it would be life-obvious for everyone.

Reviewer 3: >Read how to [list all available tags](https://github.com/management/configuration-api/v2.0#list-all-tags) and [create a new tag](https://github.com/management/configuration-api/v2.0/#add-a-tag).

@ale-kacperczyk this is exactly what I had in mind 😉 
and my remark about obviousness is not about putting the above info in but about the "the tag needs to be created first" sentence... like duh

Reviewer: OK, I get what you mean 👍🏻

> The **tagged threads** can be untagged using [Untag Thread](#untag-thread).

Reviewer 3: I don't think we mention opposite methods in other places (eg remove_user in add_user description) 🤔 

anyway, if we are going to do so, let's go with "to remove an existing tag from the thread use..."
"tagged thread can be untagged" sounds... meh

Author: It is not crucial, so I will remove it :)

### Untag Thread

To make a request, you need to specify the **tag name**, the **Id of the chat** and the **Id of the thread** you want to remove a **tag** from. Active and inactive chats can be untagged, and one or more tags can be removed from the specific **thread**. Tags are case sensitive.

> The **untagged threads** can be tagged using [Tag Thread](/messaging/agent-chat-api#tag-thread).

Reviewer: I would apply the same suggestions as for the Tag Thread method.

### Untag Thread

Untags a given thread. The following restrictions apply:

- Active and inactive chats can be untagged. 
- One or more tags can be removed from the specific **thread**. 

> - Tags are case sensitive.

Reviewer 2: Hmm, I'm thinking if this belongs here, as we know already that tags are case sensitive from the previous "Tag thread" description and it's more of a tag-specific thing rather than method-specific information 🤔

Author: I put that rather 'just to be sure', but now that I am thinking about what you have written, I am more for removing it from here.

Reviewer: Removing it from this description would assume someone has written the Tag Thread method description. I they haven't and we remove the description, one might not one that calling Untag Thread with `support` and `Support` is removing two separate tags.

Author: Good point, so this should definitely stay.
